### PR TITLE
Add gosimple linter

### PR DESCRIPTION
**Pull request by dnephin**
And update gometalinter

### CODE REVIEW

File: daemon/graphdriver/aufs/aufs_test.go [hunk 1]
```diff
@@ -179,9 +181,8 @@ func TestCleanupWithNoDirs(t *testing.T) {
 	d := newDriver(t)
 	defer os.RemoveAll(tmp)
 
-	if err := d.Cleanup(); err != nil {
-		t.Fatal(err)
-	}
+	err := d.Cleanup()
+	require.NoError(t, err)
```
Comment (thaJeztah) [inline daemon/graphdriver/aufs/aufs_test.go]: Could possibly even be;

```go
require.NoError(t, d.Cleanup())
```

File: daemon/graphdriver/aufs/aufs_test.go [hunk 2]
```diff
@@ -574,9 +560,8 @@ func TestStatus(t *testing.T) {
 	}
 
 	status := d.Status()
-	if status == nil || len(status) == 0 {
-		t.Fatal("Status should not be nil or empty")
-	}
+	assert.NotNil(t, status)
```
Comment (thaJeztah) [inline daemon/graphdriver/aufs/aufs_test.go]: Does `NotNil` also take `len(status) == 0` properly into account? i.e., in this example, only the second "if" is true; https://play.golang.org/p/CF37XvmSn9

```go
foo := [][2]string{}

if foo == nil {
	fmt.Println("I am nil")
}
if len(foo) == 0 {
	fmt.Println("I have zero items")
}
```

File: daemon/graphdriver/aufs/aufs_test.go [hunk 3]
```diff
@@ -677,48 +662,34 @@ func testMountMoreThan42Layers(t *testing.T, mountPath string) {
...
-	if len(files) != expected {
-		t.Errorf("Expected %d got %d", expected, len(files))
-	}
+	require.NoError(t, err)
+	assert.Len(t, files, expected)
 }
 
 func TestMountMoreThan42Layers(t *testing.T) {
-	os.RemoveAll(tmpOuter)
+	defer os.RemoveAll(tmpOuter)
```
Comment (thaJeztah) [inline daemon/graphdriver/aufs/aufs_test.go]: This changes the RemoveAll to be done _after_ `testMountMoreThan42Layers` instead of _before_ ?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Generally looks good, but left two questions 😊

File: daemon/graphdriver/aufs/aufs_test.go (hunk 3 above)
Comment (dnephin) [reply daemon/graphdriver/aufs/aufs_test.go]: Ya, this was just a bug in the test. Every other testcase in this file does this as a defer.

File: daemon/graphdriver/aufs/aufs_test.go (hunk 1 above)
Comment (dnephin) [reply daemon/graphdriver/aufs/aufs_test.go]: Could be, but I try to keep the function being tested on its own line. I think the test reads better when the function under test is separate from the assertions.

File: daemon/graphdriver/aufs/aufs_test.go (hunk 2 above)
Comment (dnephin) [reply daemon/graphdriver/aufs/aufs_test.go]: Ah, right a `nil` slice is both, but an empty slice is not nil.

I'll change this to `assert.Len()`

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

File: api/server/httputils/write_log_stream.go [hunk 4]
```diff
@@ -17,20 +17,13 @@ import (
 
 // WriteLogStream writes an encoded byte stream of log messages from the
 // messages channel, multiplexing them with a stdcopy.Writer if mux is true
-func WriteLogStream(ctx context.Context, w io.Writer, msgs <-chan *backend.LogMessage, config *types.ContainerLogsOptions, mux bool) {
+func WriteLogStream(_ context.Context, w io.Writer, msgs <-chan *backend.LogMessage, config *types.ContainerLogsOptions, mux bool) {
```
Comment (ripcurld0) [inline api/server/httputils/write_log_stream.go]: is there a reason to keep the `context.Context` argument?

Comment (dnephin) [reply api/server/httputils/write_log_stream.go]: I'm not sure. It might be fine to remove it.

Comment (vdemeester) [reply api/server/httputils/write_log_stream.go]: :thinking: might be to satisfy an interface maybe ?

**Review by vdemeester — APPROVED** [review]
LGTM 🐮

File: api/server/httputils/write_log_stream.go (hunk 4 above)
Comment (dnephin) [reply api/server/httputils/write_log_stream.go]: I didn't look into it much, we can remove it in a follow up if necessary

**Pull request merged**